internal/injector/builtin/testdata/client/mongo.go.snap

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package main
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+//line <generated>:1
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/go.mongodb.org/mongo-driver/mongo"
+)
+
+//line samples/client/mongo.go:16
+func mongoClient() {
+	ctx := context.Background()
+	opts := options.Client().ApplyURI("mongodb://localhost:27017")
+	client, err :=
+//line <generated>:1
+		func() (client *mongo.Client, err error) {
+			opts.Monitor = mongo.NewMonitor()
+			return mongo.Connect(ctx, opts)
+		}()
+//line samples/client/mongo.go:20
```

Comment:
If `opts` is necessarily created via `options.Client()`, then maybe we can instrument around `options.Client()`, adding the `.SetMonitor(mongo.NewMonitor())` call instead of doing the IIFE around the `mongo.Connect` call?

Reply:
@RomainMuller Done, I think it's the best way. I tried it previously but it wasn't working, but now I saw why: I was using `mongo.Client` instead of `options.Client`.